util/promlint/promlint.go

```diff
@@ -135,6 +136,30 @@ func lintMetricUnits(mf dto.MetricFamily) []Problem {
 	return problems
 }
 
+// lintCounter detects issues specific to counters, as well as patterns that should
+// only be used with counters.
+func lintCounter(mf dto.MetricFamily) []Problem {
+	var problems []Problem
+
+	isCounter := *mf.Type == dto.MetricType_COUNTER
+	hasTotalSuffix := strings.HasSuffix(*mf.Name, "_total")
+
+	switch {
+	case isCounter && !hasTotalSuffix:
+		problems = append(problems, Problem{
+			Metric: *mf.Name,
+			Text:   `counter metrics should have "_total" suffix`,
+		})
+	case !isCounter && hasTotalSuffix:
```

Comment:
Sure thing. I'll exempt them, and also keep that in mind for any future gauge checks, if we need those.